app/packs/controllers/dropzone_controller.js

```diff
@@ -35,6 +35,16 @@ export default class extends Controller {
     }
   }
 
+  checkForDuplicates(fileName) {
+    // Current files in dropZone
+    const fileNames = this.dropZone.files.map(file => { return file.name })
+    // Remove current fileName
+    fileNames.pop(fileName)
+    // Existing files for this Work
+    this.previewTargets.map(elem => { fileNames.push(elem.children[1].innerText.trim()) })
```

Comment:
It looks like you are doing a map, but aren't doing anything with the result. Would `forEach` be a better choice?

Reply:
Do we need line 40?  Don't each of them get added to the previewTargets?

Reply:
Instead of walking the DOM tree here, can you add a stimulus target attributes (`data-dropzone-target="fileName"`) to the fields and use `this.fileNameTargets`?

Reply:
Using new stimulus target attributes, removed dropZone in latest commit.